--- a/src/mutex/mut_win32.c
+++ b/src/mutex/mut_win32.c
@@ -424,7 +424,7 @@
 			 */
 			mutexp->nwaiters++;
 			ret = __db_win32_mutex_wait_int(env, mutex,
-			    (interlocked_val)&mutexp->sharecount);
+			    (LONG *)&mutexp->sharecount);
 			mutexp->nwaiters--;
 			if (ret != 0)
 				goto err;
@@ -588,16 +588,14 @@
 		if (F_ISSET(mutexp, DB_MUTEX_LOCKED)) {
 			F_CLR(mutexp, DB_MUTEX_LOCKED);
 			if ((ret = InterlockedExchange(
-			    (interlocked_val)(&atomic_read(
-			    &mutexp->sharecount)), 0)) !=
+			    (LONG *)&mutexp->sharecount, 0)) !=
 			    MUTEX_SHARE_ISEXCLUSIVE) {
 				ret = DB_RUNRECOVERY;
 				goto err;
 			}
 		} else {
-			if (InterlockedDecrement((interlocked_val)
-			    (&atomic_read(&mutexp->sharecount))) > 0)
+			if (InterlockedDecrement((LONG *)&mutexp->sharecount) > 0)
 				goto finish_shared;
 		}
 	} else